### PR TITLE
Add search to Resources, and filter courses by tag

### PR DESCRIPTION
**Pull request by katevoss**
Closes https://github.com/catalpainternational/asteroid/issues/104

I haven't added the debounce functionality as I'm not sure if it's necessary for search performance, what do you think?

Also I'm not sure why all the commits (and file changes) for `FT_Quick_References` are showing up in this PR 🤷‍♀️

### CODE REVIEW

**Comment by sandroroux** [comment]
@katevoss At which commit should your branch start?

File: src/riot/Screens/Resources.riot.html [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+
+                const showArticle = isQueryMatch && isTagMatch;
+
+                return showArticle;
+            },
+
+            toggleSearch() {
+                this.update({ searchOpen: !this.state.searchOpen });
+                if (this.state.searchOpen) {
+                    const query = this.$("#search");
```
[the remark this answers is not in the document]
Comment (sandroroux) [reply src/riot/Screens/Resources.riot.html:88]: Disregard the `searchBar` name if you create a `SearchBar` component.

File: src/riot/App.riot.html [hunk 2]
```diff
@@ -105,15 +105,22 @@
...
+                { title: "Case Management of COVID-19", description: "short description", tags: ["treatment"]},
+                { title: "Limited PPE in COVID-19", description: "all about protective equipment for health care providers",  tags: ["COVID-19"]},
+                { title: "Personal Protective Equipment (PPE) in COVID-19", description: "", tags: ["symptoms", "COVID-19"]},
+            ],
+
+            listOfTags: [
+                "COVID-19",
+                "Triage",
+                "symptoms",
+                "treatment",
```
[the remark this answers is not in the document]
Comment (katevoss) [reply src/riot/App.riot.html:123]: I agree, this was just the simplest and most central place to put the dummy data to pass it down.

File: src/riot/Components/Card.riot.html [hunk 3]
```diff
@@ -7,7 +7,7 @@ <h5 if={props.isComingSoon} translate>Coming Soon!</h5>
             <div if={props.isComingSoon} class="break-card-footer"></div>
             <p>{props.description}</p>
         </div>
-        <div class="card-footer">
+        <div if="{props.showFooter}" class="card-footer">
```
[the remark this answers is not in the document]
Comment (katevoss) [reply src/riot/Components/Card.riot.html:10]: If I follow your question, adding the `showFooter` attribute means that every instance of the `<Card>` now needs the `showFooter` attribute. I added it to the [`LessonPage` here](https://github.com/catalpainternational/canoe/pull/134/files#diff-f09c62f40f0d4ad5ea5e0ccd2ec747c4R95) does that answer your question?

**Comment by katevoss** [comment]
I'm converting this PR to a draft while I respond to comments and the updated design for the SearchBar.

**Comment by katevoss** [comment]
@PeteCoward That's a good question, I'll look into whether we can use css to modify the icons but I'd rather not hold on merging this PR just for that. I made a new issue for this: [use css to color SVGs](https://github.com/catalpainternational/asteroid/issues/113)

**Pull request merged**